src/config.rs

```diff
@@ -284,6 +287,119 @@ impl GlobalConfig {
             ColorChoice::Never | ColorChoice::Auto => false,
         }
     }
+
+    /// Set (overwrite) the [`FeatureFlag`] set.
+    #[inline]
+    pub fn set_feature_flags(&mut self, flags: HashSet<FeatureFlag>) -> &mut Self {
+        self.feature_flags = flags;
+        self
+    }
+
+    /// Enable a single [feature flag](FeatureFlag).
+    #[inline]
+    pub fn enable_feature_flag(&mut self, flag: FeatureFlag) -> &mut Self {
+        self.feature_flags.insert(flag);
+        self
+    }
+
+    /// Test for whether a specific feature flag is enabled.  If the flag has been
+    /// stabilized, this will always return true.
+    #[must_use]
+    #[inline]
+    pub fn feature_flag_enabled(&self, flag: FeatureFlag) -> bool {
+        flag.stable || self.feature_flags.contains(&flag)
+    }
+
+    /// Returns a set of all enabled feature flags.
+    #[must_use]
+    #[inline]
+    pub fn feature_flags(&self) -> &HashSet<FeatureFlag> {
+        &self.feature_flags
+    }
+}
+
+/// A feature flag for gating unstable `cargo-semver-checks` features.
+///
+/// ## Feature-gating code
+///
+/// To only execute a block of code when a given feature flag `flag` has been enabled,
+/// wrap the block in `if config.feature_flag_enabled(flag)`, where `config` is the
+/// program's [`GlobalConfig`].
+///
+/// ## Adding a new unstable feature flag
+///
+/// Create a new associated constant in the `impl FeatureFlag` block with an identifier,
+/// help message, and `stable: false`, and **add this constant** to the [`ALL_FLAGS`](Self::ALL_FLAGS)
+/// slice.
+///
+/// ## Stabilizing a feature flag
+///
+/// Set `stable: true` on the associated constant for that flag.  To keep the transition from unstable flag
+/// to stable feature, mark the associated constant `#[deprecated]`, but don't remove it.  This will warn downstream
+/// code and `cargo-semver-checks` binary users that the flag has been stabilized and may be removed in a
+/// future release, and hide it from `-Z help`, without breaking downstream code and binary users right
+/// when it is stabilized.
+///
+/// When you stabilize a flag and mark it as `#[deprecated]`, remove any `if` blocks that feature-gate code.
+/// Testing if a flag is enabled with [`GlobalConfig::feature_flag_enabled`] will always return
+/// `true` when a flag has been stabilized.
+///
+/// ## See also
```

Comment:
Try to keep line lengths under 100 chars whenever it's reasonably practical, just for convenience of allowing multiple editors side-by-side on a small screen. clippy won't lint for this AFAICT, and `cargo fmt` won't reformat docstrings, so this is a bit of a manual step. I have rulers set up in my editor to help me not exceed 100 chars, and I highly recommend that approach.